File: 13. Roman to Integer.cpp
class Solution {
public:
    int romanToInt(string s) {        
        char Rom_num[] = {'M', 'D', 'C', 'L', 'X', 'V', 'I'};
        int last_Rom[7];
        int res = 0;
        for (int i = 0; i < 7; i++)
            last_Rom[i] = s.find_last_of(Rom_num[i]);
        for (int i = 0; i < s.length(); i++){
            switch(s[i]){
                case 'M':
                    res += 1000;
                    break;
                case 'D':
                    res += 500;
                    break;
                case 'C':
                    if (i < last_Rom[0] || i < last_Rom[1])
                        res -= 100;
                    else
                        res += 100;
                    break;
                case 'L':
                    res += 50;
                    break;
                case 'X':
                    if (i < last_Rom[2] || i < last_Rom[3])
                        res -= 10;
                    else
                        res += 10;
                    break;
                case 'V':
                    res += 5;
                    break;
                case 'I':
                    if(i < last_Rom[4] || i < last_Rom[5])
                        res -= 1;
                    else
                        res += 1;
                    break;
                default:
                    break;
                        
            }
        }
        return res;        
    }
};

File: 5-Problem-A-Day/13. Roman to Integer.java
class Solution {
    public int romanToInt(String s) {
        int ans = 0;
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            if (c == 'M')
                ans += 1000;
            else if (c == 'D')
                ans += 500;
            else if (c == 'C') {
                if (i < s.length() - 1 && (s.charAt(i + 1) == 'D' || s.charAt(i + 1) == 'M'))
                    ans -= 100;
                else
                    ans += 100;
            } else if (c == 'L')
                ans += 50;
            else if (c == 'X') {
                if (i < s.length() - 1 && (s.charAt(i + 1) == 'L' || s.charAt(i + 1) == 'C'))
                    ans -= 10;
                else
                    ans += 10;
            } else if (c == 'V')
                ans += 5;
            else if (c == 'I') {
                if (i < s.length() - 1 && (s.charAt(i + 1) == 'V' || s.charAt(i + 1) == 'X'))
                    ans -= 1;
                else
                    ans += 1;
            }
        }
        return ans;
    }
}
